src/libstore/s3_binary_cache_store.go

```diff
@@ -0,0 +1,110 @@
+package libstore
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"io"
+	"log"
+	"net/url"
+)
+
+type S3Store struct {
+	Profile    string
+	Region     string
+	Scheme     string
+	BucketName string
+	Client     *s3.S3
+}
+
+func (s *S3Store) FileExists(path string) (bool, error) {
+
+	_, err := s.GetFile(path)
+	if err != nil {
+		return false, nil
```

Comment:
here you need to switch on the error type and only return false if the error is of type `s3.ErrCodeNoSuchKey`